compiler/rustc_data_structures/src/unord.rs

```diff
@@ -109,6 +109,13 @@ impl<T, I: Iterator<Item = T>> UnordItems<T, I> {
     pub fn collect<C: From<UnordItems<T, I>>>(self) -> C {
         self.into()
     }
+
+    #[track_caller]
+    pub fn take_sole_item(mut self) -> Option<T> {
```

Comment:
There's a very similar existing method called `get_only`.

Reply:
Renamed to match the existing `UnordSet::get_only`